### ISSUE TITLE
Score_eps in the actual implementation

### ISSUE THREAD

**Issue by nctamer** [Yujia-Yan/Skipping-The-Frame-Level#27]
Hi,
Thanks for your work! I have been trying to replicate the reported results. Since the model training has been taking too long, I wanted to explore what might be the reason. While debugging the code I found something that was unexpected.  Although the term is present in the last ISMIR paper, the [aggregated score_{\eps} from the no-event sections seems not used in the computation](https://github.com/Yujia-Yan/Transkun/blob/c5cb8370e17b4a1650b5971ba87ee4b0d208e5b6/transkun/LayersTransformer.py#L436). The comment says it is 'dummy eps score for testing', but the training actually only considers the positive region scores for backpropagation (which might be harder to optimize for). Was this the code that was used during your training? I thought this might explain why it is taking this long to replicate. This part is actually used in both evalPathSlow and the actual [evalPath](https://github.com/Yujia-Yan/Transkun/blob/c5cb8370e17b4a1650b5971ba87ee4b0d208e5b6/transkun/CRF/NeuralSemiCRFInterval.py#L540) function in training. Am I missing something? Is training faster without nullifying the score_eps? I would appreciate if you can kindly explain the code in comparison to the paper formulation.
Thank you!

**Comment by Yujia-Yan** [comment]
In ISMIR 2024 paper, score_eps is no longer needed as it has "minor impact to performance" and it's proven that the pairwise interval score alone is sufficient. For the current version, I simply reused the original semi-crf module, using a zero tensor as score_eps. Those computation involving score_eps is redundant in V2. Also there's some other places to optimize in the code. But I haven't done it.

For VIT-like architecture, training from scratch is known to converge slower. Some modifications and tricks in literature on similar architecture may improve the convergence. Also, optimizer and learning rate schedule may have a large impact on training time.
